src/types.ts

```diff
@@ -189,7 +193,7 @@ export interface ChainInfo {
 }
 
 export type NonRelayNativeInterior = {
-	X2: [{ PalletInstance: string }, { GeneralIndex: string }];
+	X2: [{ PalletInstance: string | undefined }, { GeneralIndex: string }];
```

Comment:
When will the palletInsance be `undefined`, i might have missed it in the above.

Reply:
If I recall the assets pallet doesnt exist for some chains such as collectives (https://github.com/paritytech/cumulus/blob/master/parachains/runtimes/collectives/collectives-polkadot/src/lib.rs) and we do throw an error for cases where its not found.

However, I think this is more a result of how I placed the palletId in SystemToSystem so Im going to refactor this back to be a string only.

Reply:
Awesome sounds good :) Thanks for the explanation